aten/src/ATen/native/PointwiseOps.cpp

```diff
@@ -16,7 +16,13 @@ TORCH_META_FUNC(addcmul)
  const Tensor& tensor1,
  const Tensor& tensor2,
  const Scalar& value) {
-  build_ternary_op(maybe_get_output(), self, tensor1, tensor2);
+  build(BINARY_OP_CONFIG()
+    .add_owned_output(maybe_get_output())
+    .add_owned_input(self)
+    .add_owned_input(tensor1)
+    .add_owned_input(tensor2));
+  // XXX: Should use this instead when it supports BINARY_OP_CONFIG.
```

Comment:
hmm what do you mean?

Reply:
the line below the comment is what the code used originally. this is what should be used instead because it's less error-prone and requires fewer lines of code. however, it cannot be used rn because the fix requires adding TI config options, which is what the code above the comment adds. the ideal fix would be just fixing the TI api to set the options everywhere, but mike wanted a smaller fix, which is why i went with this.

see this for more discussion https://github.com/pytorch/pytorch/issues/70486#issuecomment-1059227417

Reply:
This isn't the smaller fix I was hoping for, though. `addcmul` shouldn't be reusing `BINARY_OP_CONFIG` since it's a ternary operator.

Reply:
In particular, I would try to just extend the ternary op TensorIterator options to perform type promotion like the existing elementwise builders do.

Reply:
okay, i pushed another update. i hope i did it right this time